src/analysis/fourier/poisson_summation.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2022 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import analysis.fourier.add_circle
+import measure_theory.integral.exp_decay
+import analysis.special_functions.gaussian
+
+/-!
+# Poisson's summation formula
+
+We prove Poisson's summation formula `∑ (n : ℤ), f n = ∑ (n : ℤ), g n`, where `g` is the Fourier
+transform of `f`, under the following hypotheses:
+* `f` and `g` are continuous functions `ℝ → ℂ`.
+* `∀ t:ℝ, g t = ∫ x:ℝ, complex.exp (-2 * π * I * t * x) * f x`.
+* `f` is integrable.
+* The sum `∑ (n:ℤ), g n` is convergent.
+* For all compacts `K ⊂ ℝ`, the sum `∑ (n:ℤ), sup { ‖f(x + n)‖ | x ∈ K }` is convergent.
+
+The last three conditions follow if we assume both `f` and `g` have exponential decay -- this will
+be added in future.
+-/
+
+noncomputable theory
+
+open_locale real big_operators filter
+open function complex set topological_space filter measure_theory
+
+/-- The continuous map `λ x, f(x + m)`. -/
+def continuous_map.comp_add (f : C(ℝ, ℂ)) (m : ℝ) : C(ℝ, ℂ) :=
+continuous_map.mk (λ x:ℝ, f $ x + m) (by continuity)
```

Comment:
I think this would have the undesirable side-effect (for this particular code) of requiring an explicit type cast if `m` is an integer (or something else which isn't in `ℝ`), because `continuous_map.add_right m` would be an element of `C(ℤ, ℤ)` (not a useful object here).

Reply:
I suspect you only need to write an explicit `\u`, rather than actually writing the destination type.